Build failure: freeswitch

### Nixpkgs version

- Stable (25.05)

### Steps to reproduce

Try to build the package. 

### Can Hydra reproduce this build failure?

Yes, Hydra can reproduce this build failure.

### Link to Hydra build job

https://hydra.nixos.org/build/298608796

### Relevant log output

```console
make[4]: Entering directory '/build/source/src/mod/xml_int/mod_xml_rpc'
make[4]: warning: -j1 forced in makefile: resetting jobserver mode.
  CC       mod_xml_rpc_la-mod_xml_rpc.lo
make[4]: Leaving directory '/build/source/src/mod/applications/mod_voicemail'

making all mod_xml_scgi
make[4]: Entering directory '/build/source/src/mod/xml_int/mod_xml_scgi'
  CC       mod_xml_scgi_la-mod_xml_scgi.lo
mod_xml_rpc.c: In function 'is_authorized':
mod_xml_rpc.c:362:67: error: passing argument 6 of 'user_attributes' from incompatible pointer type []
  362 |         if (!user_attributes(user, domain_name, NULL, NULL, NULL, &allowed_commands)) {
      |                                                                   ^~~~~~~~~~~~~~~~~
      |                                                                   |
      |                                                                   char **
mod_xml_rpc.c:249:148: note: expected 'const char **' but argument is of type 'char **'
  249 |                                                                   const char **ppasswd, const char **pvm_passwd, const char **palias, const char **pallowed_commands)
      |                                                                                                                                       ~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~
mod_xml_rpc.c: In function 'handler_hook':
mod_xml_rpc.c:947:86: error: passing argument 5 of 'ConnRead' from incompatible pointer type []
  947 |                                                 ConnRead(r->connP, 2000, NULL, NULL, &readError);
      |                                                                                      ^~~~~~~~~~
      |                                                                                      |
      |                                                                                      char **
In file included from /build/source/libs/xmlrpc-c/include/../lib/abyss/src/http.h:7,
                 from mod_xml_rpc.c:70:
/build/source/libs/xmlrpc-c/include/../lib/abyss/src/conn.h:98:30: note: expected 'const char ** const' but argument is of type 'char **'
   98 |          const char ** const errorP);
      |          ~~~~~~~~~~~~~~~~~~~~^~~~~~
make[4]: *** [Makefile:1294: mod_xml_rpc_la-mod_xml_rpc.lo] Error 1
make[4]: Leaving directory '/build/source/src/mod/xml_int/mod_xml_rpc'
```

### Additional context

_No response_

### System metadata

 - system: `"x86_64-linux"`
 - host os: `Linux 6.6.87, NixOS, 24.11 (Vicuna), 24.11.20250412.26d499f`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.24.14`
 - nixpkgs: `/nix/store/hzaj4d6ari2wq2cbg1j60n9zw42gnshy-source`

### Notify maintainers

@MikaelFangel 

---

**Note for maintainers:** Please tag this issue in your pull request description. (i.e. `Resolves #ISSUE`.)


### I assert that this issue is relevant for Nixpkgs

- [x] I assert that this is a bug and not a support request.
- [x] I assert that this is not a [duplicate of an existing issue](https://github.com/NixOS/nixpkgs/issues?q=is%3Aopen+is%3Aissue+label%3A%220.kind%3A+build+failure%22). 
- [x] I assert that I have read the [NixOS Code of Conduct](https://github.com/NixOS/.github/blob/master/CODE_OF_CONDUCT.md) and agree to abide by it.

### Is this issue important to you?

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/by-name/fr/freeswitch/package.nix
+++ b/pkgs/by-name/fr/freeswitch/package.nix
@@ -158,7 +158,11 @@ stdenv.mkDerivation rec {
 
   enableParallelBuilding = true;
 
-  env.NIX_CFLAGS_COMPILE = "-Wno-error";
+  env.NIX_CFLAGS_COMPILE = toString [
+    "-Wno-error"
+    # https://github.com/signalwire/freeswitch/issues/2495
+    "-Wno-incompatible-pointer-types"
+  ];
 
   # Using c++14 because of build error
   # gsm_at.h:94:32: error: ISO C++17 does not allow dynamic exception specifications

